src/main/common/utils.js

```diff
@@ -188,6 +188,78 @@ gpii.app.isPointInRect = function (point, rectangle) {
            rectangle.y <= point.y && point.y <= rectangle.y + rectangle.height;
 };
 
+/**
+ * Checks if the buttonList attribute exists in the siteConfig object
+ * @param {Object} siteConfig - instance of the siteConfig object
+ * @returns {Boolean}
+ */
+gpii.app.isButtonList = function(siteConfig) {
+    return fluid.isValue(siteConfig.buttonList);
+};
+
+/**
+ * Looks for a `id` and matches it to the provided string
+ * return empty array when there is no button found
+ * @param {String} buttonId - the `id` of the button
+ * @param {Array} availableButtons - the full settings list
+ * @returns {Array}
+ */
+gpii.app.findButtonId = function(buttonId, availableButtons) {
+    var returnData = [];  // returning empty array by default
+
+    if (fluid.isValue(buttonId) && fluid.isValue(availableButtons)) {
+        fluid.each(availableButtons, function(button) {
+            if (fluid.isValue(button.id)) {
+                if (button.id === buttonId) {
+                    returnData = button; // returning the button data
+                }
+            }
+        });
+    }
+    return returnData;
+};
+
+/**
+ * Filters the full button list based on the provided array of `id` attributes
+ * @param {Array} siteConfigButtonList - basic array of strings
+ * @param {Object[]} availableButtons - all available buttons found in settings.json
+ * @returns {Object[]} - filtered version of available buttons (same structure)
+ */
+gpii.app.filterButtonList = function(siteConfigButtonList, availableButtons) {
+    var matchedList = [], // these buttons are explicitly selected in the
+                          // siteConfig, added in the same order
+        afterList = [],   // all the buttons that don't have `id` at all,
+                          // they are added at the end of the list
+        tabindex = 100;   // starting tabindex, adding +10 of each new item
+
+    // creating the matchedList
+    // looking for `id` and if matches adding it
+    fluid.each(siteConfigButtonList, function(buttonId) {
+        var matchedButton = gpii.app.findButtonId(buttonId, availableButtons);
+        if (matchedButton !== []) {
```

Comment:
I feel sure that if test cases were written for this functionality, one would find that this comparison does not work! Since [] !== []

Reply:
True, checking if its not false now (based on the change above).